### ISSUE TITLE
[7.0.0] Deprecated warning from immutablejs (iterable.length)

### ISSUE THREAD

**Issue by giautm** [erikras/redux-form#3196]
### **bug report**
Deprecated warning from immutablejs

### What is the current behavior?

Show Deprecated warning from immutablejs:
> iterable.length has been deprecated, use iterable.size or iterable.count()

### What is the expected behavior?
Don't show up warning.

### What's your environment?
```
"react": "16.0.0-alpha.12",
"redux": "^3.6.0",
"redux-form": "^7.0.0",
"immutable": "3.7.6",
```
### Other information
![image](https://user-images.githubusercontent.com/12751435/28296814-a9e00dde-6b94-11e7-9234-13586610307f.png)

**Comment by erikras** [comment]
I went to test this, and installed the latest version of `immutable`, which is `3.8.1`, and I got no stack trace. So I tried your specific version, `3.7.6`, and saw that the error says, **This warning will become a silent error in a future version**. 🤦‍♂️ What sense does that make? Is it an error or not? Seems like questionable library management, but perhaps they have their reasons.

Regardless, `redux-form` should not be doing a plain `deepEqual` check on an immutable `Map`, so this really is a bug.

**Comment by erikras** [comment]
Okay, my previous conclusion was not entirely correct. Sometimes this error is being caused by lodash's `deepEqual` checking if the value is "array like", by seeing if it has a numeric `.length` attribute. That hardly seems like a sin worth throwing up a warning for. As such, it _does_ sort of make sense that immutable.js went through one version where it warned everybody who might have been checking `.length` the old way, but that it _should_ "fail" silently in future versions.

I did find one bug in my investigation, so that's good. However, lodash is certainly not committing any crime by probing an attribute on a value.

### TL;DR:

This is not really a problem. Upgrade to immutable.js `3.8.x`.

**Comment by erikras** [comment]
Semi-fix published in [`v7.0.1`](https://github.com/erikras/redux-form/releases/tag/v7.0.1).

**Comment by giautm** [comment]
So fast. 😄
